modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java

```diff
@@ -249,18 +262,23 @@ private static int getLogFileSequenceNumber() {
                         + "found " + attributeExpressionExecutors[1].getReturnType());
             }
 
-            if (attributeExpressionLength == 3) {
-                if (attributeExpressionExecutors[2].getReturnType() == Attribute.Type.INT) {
+            if (!(attributeExpressionExecutors[2].getReturnType() == Attribute.Type.INT)) {
```

Comment:
Shouldn't this attribute be checked for whether it is a variable or constant? I see you have done that checking for the first two attributes but not for this attribute.

Reply:
Modified